### PR TITLE
MFA Example

### PR DESCRIPTION
**Pull request by awhite-transmit**
Example following [https://transmitsecurity.atlassian.net/browse/DEVX-23](https://transmitsecurity.atlassian.net/browse/DEVX-23) with a change in that once the user is created they need to login (the password example that this draws from auto logs them in).  This was done to make things consistent when running the example with a new user that is created or an existing user that was created outside the example.

### CODE REVIEW

**Comment by ts-ron** [comment]
@awhite-transmit overall looks great and I am still reviewing.
Initial registration and authentication works well, however I am getting the following error when I try to login for the 2nd time:
image


are you familiar with it?

**Comment by awhite-transmit** [comment]
> @awhite-transmit overall looks great and I am still reviewing. Initial registration and authentication works well, however I am getting the following error when I try to login for the 2nd time: image
> 
> are you familiar with it? When I go to the portal the user is marked as "Pending" and not "Active" - maybe some missing API call during the registration flow?

@ts-ron This error occurs because the platform maintains a session for the user in the browser and because of that thinks you have already authenticated with password and therefore that method is unavailable to use for MFA.  I call out in the readme that the example needs to be run in a new private browser session EACH time, otherwise you will encounter it there as well on the second run through it.

As far as the pending status, I have always seen this, I think the only time I saw it go away was when I authenticated a user using hosted webauthn.  If I am missing some call/attribute during creation let me know, I will check again but not aware of anything, this just seems to be the behavior...

**Comment by awhite-transmit** [comment]
@ts-ron I added logout to the restart button, if the user follows that flow it will work, still possible to get in the existing session state for instance if the app restarts (they are actively editing it) but this should resolve it for the expected flow scenario.

**Comment by ts-ron** [comment]
> @ts-ron I added logout to the restart button, if the user follows that flow it will work. It is still possible to get in the existing session state (where mfa breaks) for instance if the app restarts (like they are actively editing it) but this should resolve it for the expected flow scenario. It also turns out that by logging out the user their status within the admin portal changes to "Active".

@awhite-transmit perfect :)

File: login-with-mfa/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  (https://developer.transmitsecurity.com/guides/user/auth_mfa_guide/#step-1-request-mfa)
+- This will cause a redirect to the /complete endpoint with an error indicating MFA is required
+  (https://developer.transmitsecurity.com/guides/user/auth_mfa_guide/#step-2-handle-mfa-error)
+- An OTP code will be sent to the phone number you signed up with
+  (https://developer.transmitsecurity.com/guides/user/auth_sms_otp/)
+- Check your phone for the OTP code sent through SMS, enter the code and click the Login button
+- The browser will redirect to the completion endpoint and show the OIDC auth code now that the MFA
+  flow is complete
+- It is recommended to open the browser debugger tools and look at the console logs which reflect
+  flow and network. It is also strongly recommended to run this flow in a **new** Private or
```
Comment (ts-ron) [inline login-with-mfa/README.md:35]: @awhite-transmit before merging - maybe change the recommendation from using private/incognito windows to using "restart" to logout, and not going back to the first page?

Comment (awhite-transmit) [reply login-with-mfa/README.md:35]: @ts-ron ahh, too late!  don't think I can edit stuff in main, will create another pull request to fix it.

**Pull request merged**